llms/mlx_lm/tuner/dora.py

```diff
@@ -0,0 +1,119 @@
+import math
+from typing import Any
+
+import mlx.core as mx
+import mlx.nn as nn
+
+
+
+class DoRALinear(nn.Module):
+    @staticmethod
+    def from_linear(
+        linear: nn.Linear,
+        r: int = 8,
+        alpha: float = 16,
+        dropout: float = 0.0,
+        scale: float = 10.0,
+    ):
+        # TODO remove when input_dims and output_dims are attributes
+        # on linear and quantized linear
+        output_dims, input_dims = linear.weight.shape
+        if isinstance(linear, nn.QuantizedLinear):
+            input_dims *= 32 // linear.bits
```

Comment:
```suggestion
        if isinstance(linear, nn.QuantizedLinear):
            raise ValueError("DoRALinear does not yet support quantized weights.")
```

Reply:
remove  `input_dims *= 32 // linear.bits` and add `raise ValueError("DoRALinear does not yet support quantized weights.")`